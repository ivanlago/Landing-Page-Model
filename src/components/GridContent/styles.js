import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Hmtl = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
