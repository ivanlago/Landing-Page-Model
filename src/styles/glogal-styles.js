import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

html {
  font-size: 62.5%;
  scroll-behavior: smooth;
}

body {
  ${({ theme }) => css`
    background: ${theme.colors.white};
  `}
  font-size: 1.6rem;
  font-family: 'Open Sans', sans-serif;
}

h1, h2, h3, h4, h5, h6 {
  font-family: 'Montserrat', sans-serif;
}

p {
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
  `}
}

ul, ol {
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    padding: ${theme.spacings.medium};
  `}
}

`;
