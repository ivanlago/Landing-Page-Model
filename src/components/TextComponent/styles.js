import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
