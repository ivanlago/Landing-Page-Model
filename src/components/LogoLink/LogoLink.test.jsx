import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="Ola Mundo" />);
    const heading = screen.getByRole('heading', { name: 'Ola Mundo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });
  it('should render image logo', () => {
    renderTheme(<LogoLink link="#target" text="Ola Mundo" srcImg="image.jpg" />);
    const heading = screen.getByRole('heading', { name: 'Ola Mundo' });
    expect(heading.firstChild.firstChild).toHaveAttribute('src', 'image.jpg');
  });
  it('should render snapshot', () => {
    const { container } = renderTheme(<LogoLink link="#target" text="Ola Mundo" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
