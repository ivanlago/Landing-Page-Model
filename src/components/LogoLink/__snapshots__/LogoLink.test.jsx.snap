// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should render snapshot 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > img {
  max-height: 4rem;
}

.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    Ola Mundo
  </a>
</h1>
`;
